Graphics/Sudbury/WirePackages.hs

```diff
@@ -26,37 +24,33 @@ data WirePackage = WirePackage
   , wirePackagePayload :: B.ByteString
   } deriving (Eq, Show)
 
--- | Construct a wayland wire wirePackage
-wirePack :: WirePackage -> BB.Builder
-wirePack msg = wirePackBuilder
-  (wirePackageSender msg)
-  (wirePackageSize msg)
-  (wirePackageOpcode msg)
-  (BB.byteString $ wirePackagePayload msg)
-
--- | Construct a wayland wire wirePackage with payload given as a 'Builder'.
-wirePackBuilder :: Word32 -> Word16 -> Word16 -> BB.Builder -> BB.Builder
-wirePackBuilder sender size opcode payload =
-  BBE.word32Host sender
-  <>
-  -- FIXME make byte order portable here
-  BBE.word16Host opcode
-  <>
-  BBE.word16Host size
-  <>
-  payload
+-- Storable that writes in the Wayland binary format.
+-- Allows us to use the optimized machinery for Data.Store
+-- to gain faster performing decoding/encoding.
+instance Store WirePackage where
+  size = VarSize $ fromIntegral . wirePackageSize
+  poke p = do
+    poke $ wirePackageSender p
+    poke $ wirePackageOpcode p 
+    poke $ wirePackageSize p
+    let (sourceFp, sourceOffset, sourceLength) = B.toForeignPtr $ wirePackagePayload p
+    pokeFromForeignPtr sourceFp sourceOffset sourceLength
+  peek = do
+    sen <- peek
+    op <- peek
+    sz <- peek
+    let payloadSize = fromIntegral sz - 8
+    plraw <- peekToPlainForeignPtr "Data.ByteString.ByteString" payloadSize
+    let pl = B.PS plraw 0 payloadSize
+    return $ WirePackage 
+      { wirePackageSender=sen
+      , wirePackageSize=sz 
+      , wirePackageOpcode=op 
+      , wirePackagePayload=pl
+      }
+  {-# INLINE size #-}
+  {-# INLINE peek #-}
+  {-# INLINE poke #-}
```

Comment:
I was going to complain about these pragmas, but store itself seems to have them as well, so I suppose we should add a comment saying that we just follow those conventions.
Also, perhaps future readers might be confused for a second why we can't just `peek` the payload. So perhaps a comment saying they're variable size.
(There seems to be support in store for `ByteString`s using some `StaticSize` type, but that's probably overkill for us.)